test/src/test/java/org/corfudb/integration/CorfuReplicationClusterConfigIT.java

```diff
@@ -800,6 +811,412 @@ public static void checkpointAndTrimCorfuStore(CorfuRuntime cpRuntime) {
+            txnContext.commit();
+        }
+
+        // Release Source's lock by deleting the lock table
+        clearLockTable(false);
+        log.info("Source's lock table cleared!");
+
+        countDownLatch.await();
+
+        Table<LogReplicationSession, LogReplicationMetadata.ReplicationMetadata, Message> metadataTable =
+                sourceCorfuStore.openTable(LogReplicationMetadataManager.NAMESPACE,
+                        REPLICATION_STATUS_TABLE,
+                        LogReplicationSession.class,
+                        LogReplicationMetadata.ReplicationMetadata.class,
+                        null,
+                        TableOptions.fromProtoSchema(ReplicationStatus.class));
+
+        try(TxnContext txn = sourceCorfuStore.txn(LogReplicationMetadataManager.NAMESPACE)) {
+            txn.delete( sourceStatusTable, sessionKey);
+            txn.delete(metadataTable, sessionKey);
+            txn.commit();
+        }
+
+        // Perform a config update with all sink
+        try (TxnContext txn = sourceCorfuStore.txn(DefaultClusterManager.CONFIG_NAMESPACE)) {
+            txn.putRecord(configTable, DefaultClusterManager.OP_ALL_SINK, DefaultClusterManager.OP_ALL_SINK, DefaultClusterManager.OP_ALL_SINK);
+            txn.commit();
+        }
+        assertThat(configTable.count()).isOne();
+        log.info("New topology config applied! All the clsuters are sink");
+        TimeUnit.SECONDS.sleep(mediumInterval);
+
+        log.info("Adding 2nd batch");
+
+        for (int i = firstBatch; i < secondBatch; i++) {
+            try (TxnContext txn = sourceCorfuStore.txn(NAMESPACE)) {
+                txn.putRecord(mapSource, StringKey.newBuilder().setKey(String.valueOf(i)).build(),
+                        IntValue.newBuilder().setValue(i).build(), null);
+                txn.commit();
+            }
+        }
+        assertThat(mapSource.count()).isEqualTo(secondBatch);
+
+        try (TxnContext txnContext = sinkCorfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            LockDataTypes.LockData oldLockData = LockDataTypes.LockData.newBuilder().mergeFrom(lockTableRecord.getPayload()).build();
+            txnContext.putRecord(sinkLockTable, lockId,
+                    LockDataTypes.LockData.newBuilder().mergeFrom(oldLockData)
+                            .setLeaseAcquisitionNumber(oldLockData.getLeaseAcquisitionNumber() + 1)
+                            .setLeaseRenewalNumber(oldLockData.getLeaseRenewalNumber() + 1).build(), null);
+            txnContext.commit();
+        }
+
+        Assert.assertEquals(1, sinkLockTable.count());
+        log.info("Source's lock table has a lock entry");
+
+        assertThat(mapSink.count()).isEqualTo(firstBatch);
+
+
+        clearLockTable(false);
+
+//        Table<LogReplicationSession, LogReplicationMetadata.ReplicationMetadata, Message> metadataTable =
+//                sinkCorfuStore.openTable(LogReplicationMetadataManager.NAMESPACE,
```

Comment:
we can remove this table open right?

Reply:
Yup, Had removed this comment in the last commit.
In the recent commit, I have addressed the codacy errors which can be addressed.

Reply:
Most of Codacy issues are not coming from the scoped transaction CL, but shows up as a result of code restructuring that was originally there.